solidago/src/solidago/solvers/dichotomy.py

```diff
@@ -29,7 +30,7 @@ def solve(
     -------
     out: float
     """
-    ymin, ymax = f(xmin) - value, f(xmax) - value
+    ymin, ymax = f(xmin, *args) - value, f(xmax, *args) - value
```

Comment:
minor: Another way to do something similar would be to not change solve, but use it with a `partial` https://docs.python.org/3/library/functools.html#functools.partial